src/virtual_elements.ts

```diff
@@ -129,12 +129,13 @@ function elementOpen(
     applyStatics(node, data, statics);
   }
 
-  if (arguments.length <= ATTRIBUTES_OFFSET && !data.hasAttrsArr()) {
+  const attrsLength = Math.max(0, arguments.length - ATTRIBUTES_OFFSET);
+  if (!attrsLength && !data.hasAttrsArr()) {
     return node;
   }
 
-  const attrsArr = data.getAttrsArr();
-  const isNew = !attrsArr.length;
+  const isNew = data.hasEmptyAttrsArr();
```

Comment:
Should we rename isNew? It could be an existing Element that had no attributes before, but has some now.

Reply:
Used your `hadNoAttrs`.